src/mbgl/util/mapbox.cpp

```diff
@@ -44,7 +44,16 @@ std::string normalizeSourceURL(const std::string& url, const std::string& access
         throw std::runtime_error("You must provide a Mapbox API access token for Mapbox tile sources");
     }
 
-    return baseURL + "v4/" + url.substr(protocol.length()) + ".json?access_token=" + accessToken + "&secure";
+    auto queryIdx = url.find("?");
+    std::string query;
+    if (queryIdx != std::string::npos) {
+        query = url.substr(queryIdx + 1, url.length() - queryIdx + 1);
+        if (query.length() > 0) {
```

Comment:
Generally `if (!query.empty()) {` is better practice since it is faster than `query.size()`.